cl/api/views.py

```diff
@@ -164,16 +164,20 @@ def coverage_data(request, version, court):
     )
 
 
-def fetch_start_end_dates_for_court(court_id: str):
-    """Fetch date and jurisdiction data for a court
+def fetch_first_last_date_filed(
+    court_id: str,
+) -> tuple[Optional[date], Optional[date]]:
+    """Fetch first and last date for court
 
     :param court_id: Court object id
-    :return: date and jurisdiction data for a particular court
+    :return: First/last date filed, if any
     """
     query = OpinionCluster.objects.filter(docket__court=court_id).order_by(
         "date_filed"
     )
-    return query.first(), query.last()
+    if query:
+        return query.first().date_filed, query.last().date_filed
```

Comment:
I'm sorry this area is performance-sensitive, so we need to be a bit careful. 

I _think_ the way you have this will run three queries:

1. `SELECT * FROM search_opinioncluster ORDER BY XYZ ASC LIMIT 20`
2. `SELECT * FROM search_opinioncluster ORDER BY XYZ ASC LIMIT 1`
3. `SELECT * FROM search_opinioncluster ORDER BY XYZ DESC LIMIT 1`

It's possible it only runs query 1 and 3, if Django is smart (it often is!). But what you really want it to run is numbers 2 and 3, because grabbing 20 results is slower than grabbing one.

Sooooo....I think you just need to handle this via a try/except instead of an `if query`?

Reply:
how about this instead

```
query = OpinionCluster.objects.filter(docket__court=court_id).order_by(
        "date_filed"
    )
first, last = query.first(), query.last()
if first:
    return first.date_filed, last.date_filed
```

Reply:
calling .first() on an empty query set isnt going to fail - but it should be false-y I think.

Reply:
Sure, that seems like it'd do it.